src/invocation/InvocationService.ts

```diff
@@ -133,13 +140,13 @@ export class InvocationService {
     private invokeSmart(invocation: Invocation) {
         if (invocation.hasOwnProperty('connection')) {
             return this.send(invocation, invocation.connection);
-        } else if (invocation.hasOwnProperty('partitionId')) {
-            var address = this.client.getPartitionService().getAddressForPartition(invocation.partitionId);
+        } else if (invocation.hasPartitionId()) {
+            const address = this.client.getPartitionService().getAddressForPartition(invocation.partitionId);
             return this.sendToAddress(invocation, address);
         } else if (invocation.hasOwnProperty('address')) {
             return this.sendToAddress(invocation, invocation.address);
         } else {
-            return this.send(invocation, this.client.getClusterService().getOwnerConnection());
+            return this.sendToAddress(invocation, this.client.getClusterService().getRandomAddress());
```

Comment:
We deliberately send the message to the owner connection if there are no other bounds. It should not be random.
Also, this refactoring is out of this pr's scope.